### PR TITLE
[OTHER] <Click link click increase tests> /tests/profile.spec.js 

### PR DESCRIPTION
**Pull request by kbventures**
## Fixes Issue
Tests that confirms that the click count in profile is accurate and working on the browser. 

Closes [OTHER] <Click link click increase tests> /tests/profile.spec.js #9384 #9537
https://github.com/EddieHubCommunity/BioDrop/issues/9537

## Changes proposed

```
import { Profile, Link } from "@models/index";
test("Link clicks increase", async ({page}) => {
  await connectMongo();
  await page.goto("/eddiejaoude");
  
  const startingLink = await Link.findOne({  url: "https://twitter.com/eddiejaoude" })

  const previousClickCount = startingLink.clicks; 
  console.log("previousClickCount", previousClickCount)

  const profileLink = page.locator('a').filter({ hasText: 'Twitter: Follow me' })

  await profileLink.click();
  await page.waitForTimeout(1000);
  const currentLink = await Link.findOne({ url: "https://twitter.com/eddiejaoude" })

  expect(currentLink.clicks).toEqual((previousClickCount + 1));
});
```

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

File: tests/profile.spec.js [hunk 1]
```diff
@@ -155,3 +155,21 @@ test.describe("accessibility tests (dark)", () => {
...
 });
+
+test("Link clicks increase", async ({page}) => {
+  await connectMongo();
+  await page.goto("/eddiejaoude");
+  
+  const startingLink = await Link.findOne({  url: "https://twitter.com/eddiejaoude" })
+
+  const previousClickCount = startingLink.clicks; 
+  console.log("previousClickCount", previousClickCount)
```
Comment (eddiejaoude) [inline tests/profile.spec.js:166]: Is the `console.log` still needed?

File: tests/profile.spec.js [hunk 2]
```diff
@@ -155,3 +155,21 @@ test.describe("accessibility tests (dark)", () => {
     expect(accessibilityScanResults.violations).toEqual([]);
   });
 });
+
+test("Link clicks increase", async ({page}) => {
+  await connectMongo();
+  await page.goto("/eddiejaoude");
+  
+  const startingLink = await Link.findOne({  url: "https://twitter.com/eddiejaoude" })
```
Comment (eddiejaoude) [inline tests/profile.spec.js:163]: I would recommend using the username also here to avoid any confusion in the test data potentially later

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Looks good 👍 

I left a few small comments

File: tests/profile.spec.js (hunk 1 above)
Comment (kbventures) [reply tests/profile.spec.js]: Removed.

File: tests/profile.spec.js [hunk 3]
```diff
@@ -155,3 +155,21 @@ test.describe("accessibility tests (dark)", () => {
     expect(accessibilityScanResults.violations).toEqual([]);
   });
 });
+
+test("Link clicks increase", async ({page}) => {
```
[the remark this answers is not in the document]
Comment (kbventures) [reply tests/profile.spec.js]: Duplicate test has been deleted and new tests moved to line 72.

File: tests/profile.spec.js (hunk 2 above)
Comment (kbventures) [reply tests/profile.spec.js]: `test("Link clicks increase", async ({page}) => {

  await connectMongo();
  await page.goto("/eddiejaoude");
  
  const startingLinks = await Link.find({ username: "eddiejaoude" })
  const startingLink = startingLinks[0]

  const previousClickCount = startingLink.clicks; 

  const profileLink = page.locator('a').filter({ hasText: startingLink.name })

  await profileLink.click();
  await page.waitForTimeout(1000);
  const currentLink = await Link.findOne({ name: startingLink.name, username: "eddiejaoude" })
  const updateCurrentLinkClicks = currentLink.clicks
  
  expect(updateCurrentLinkClicks).toEqual((previousClickCount + 1));
});`

Comment (kbventures) [reply tests/profile.spec.js]: This version is more dynamic/flexible.

Comment (eddiejaoude) [reply tests/profile.spec.js]: This looks great 👍

**Review by eddiejaoude — APPROVED** [review]

**Review by eddiejaoude — APPROVED** [review]
Looks great 👍

**Pull request merged**